src/SlnParser/Helper/EnrichSolutionWithSolutionGuid.cs

```diff
@@ -0,0 +1,38 @@
+﻿using SlnParser.Contracts;
+using SlnParser.Contracts.Helper;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace SlnParser.Helper
+{
+    internal sealed class EnrichSolutionWithSolutionGuid : IEnrichSolution
+    {
+        public void Enrich(Solution solution, IEnumerable<string> fileContents)
+        {
+            var extensibilityGlobals = SectionParser.GetFileContentsInGlobalSection(
+                fileContents,
+                "ExtensibilityGlobals");
+
+            IEnumerable<Guid?> solutionGuids = extensibilityGlobals.Select(ExtractSolutionGuid).Where(x => x.HasValue);
+            if (solutionGuids.Any())
+            {
+                solution.Guid = solutionGuids.Single();
+            }
```

Comment:
This call can be simplified to

```cs
solution.Guid = extensibilityGlobals
  .Select(ExtractSolutionGuid)
  .FirstOrDefault(x => x.HasValue);
```

This will also fix the issue that `Any()` and `Single()` would enumerate the list multiple times, which needs to be avoided.